presto-tests/src/test/java/io/prestosql/tests/TestProcedureCreation.java

```diff
@@ -0,0 +1,197 @@
+                () -> createTestProcedure(ImmutableList.of(
+                        new Procedure.Argument("name", VARCHAR, true, null),
+                        new Procedure.Argument("name2", VARCHAR, true, null),
+                        new Procedure.Argument("name3", VARCHAR, true, null),
+                        new Procedure.Argument("name4", VARCHAR, false, null),
+                        new Procedure.Argument("name5", VARCHAR, true, null))));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenArgumentNameRepeates()
+    {
+        assertThrows(
+                IllegalArgumentException.class,
+                "Duplicate argument name: 'name'",
+                () -> createTestProcedure(ImmutableList.of(
+                        new Procedure.Argument("name", VARCHAR, false, null),
+                        new Procedure.Argument("name", VARCHAR, true, null))));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenProcedureIsNonVoid()
+    {
+        assertThrows(
+                IllegalArgumentException.class,
+                "Method must return void",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(),
+                        MethodHandleUtil.methodHandle(Procedures.class, "funWithoutArguments")));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenMethodHandleIsNull()
+    {
+        assertThrows(
+                NullPointerException.class,
+                "methodHandle is null",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(),
+                        null));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenMethodHandleHasVarargs()
+    {
+        assertThrows(
+                IllegalArgumentException.class,
+                "Method must have fixed arity",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(),
+                        MethodHandleUtil.methodHandle(Procedures.class, "funWithVarargs", String[].class)));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenArgumentCountDoesntMatch()
+    {
+        assertThrows(
+                IllegalArgumentException.class,
+                "Method parameter count must match arguments",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(
+                                new Procedure.Argument("name", VARCHAR, true, null),
+                                new Procedure.Argument("name2", VARCHAR, true, null),
+                                new Procedure.Argument("name3", VARCHAR, true, null)),
+                        MethodHandleUtil.methodHandle(Procedures.class, "fun1", ConnectorSession.class, Object.class)));
```

Comment:
Static import `methodHandle` as the name stands by itself (the `MethodHandleUtil` prefix doesn't provide any value and actually detracts from readability)